File: app/src/main/java/com/flysolo/dmmsugradelevelapp/services/auth/AuthService.java
package com.flysolo.dmmsugradelevelapp.services.auth;

import android.accounts.Account;
import android.net.Uri;

import com.flysolo.dmmsugradelevelapp.model.Accounts;
import com.flysolo.dmmsugradelevelapp.utils.UiState;
import com.google.firebase.auth.FirebaseUser;

import java.net.URI;
import java.util.List;

public interface AuthService {
        void login(String email , String password, UiState<FirebaseUser> result);
        void signup(String email, String password, Accounts accounts, UiState<Accounts> result);
        void createAccount(Accounts accounts,UiState<String> result);
        void getAccount(String uid,UiState<Accounts> result);
        void reAuthenticateAccount( FirebaseUser user,String email ,String password,UiState<FirebaseUser> result);
        void resetPassword(String email,UiState<String> result);
        void changePassword(FirebaseUser user,String password,UiState<String> result);
        void uploadProfile(String uid , Uri uri,String type, UiState<String> result);
        void updateProfile(String uid,String name ,String profile,UiState<String> result);
        void getAllStudentAccount(UiState<List<Accounts>> result);
}

File: app/src/main/java/com/flysolo/myapplication/services/auth/AuthService.kt
package com.flysolo.myapplication.services.auth

import android.net.Uri

import com.flysolo.myapplication.utils.UiState
import com.flysolo.myapplication.models.Accounts
import com.google.firebase.auth.FirebaseUser

interface AuthService {
    fun login(email : String,password : String,result  : (UiState<FirebaseUser>) -> Unit)
    fun signup(email: String,password: String,user : Accounts,result  : (UiState<Accounts>) -> Unit)
    fun createAccount(user : Accounts,result  : (UiState<Boolean>) -> Unit)
    fun getUserInfo(id : String ,result  : (UiState<Accounts>) -> Unit)
    fun reAuthenticateAccount( user : FirebaseUser,email: String,password: String,result: (UiState<FirebaseUser>) -> Unit)
    fun resetPassword(email : String,result: (UiState<String>) -> Unit)
    fun changePassword(user: FirebaseUser,password: String,result: (UiState<Boolean>) -> Unit)
    fun updateUserName(uid : String , name : String,result: (UiState<String>) -> Unit)
    fun logout()
    fun getAllStudent(result: (UiState<List<Accounts>>) -> Unit)
    fun updateAvatar(uid : String,avatar : String ,result: (UiState<String>) -> Unit)
    fun updateInfo(uid: String,name : String,gender : String,result: (UiState<String>) -> Unit)
    fun uploadProfile(uid: String,uri: Uri,result: (UiState<String>) -> Unit)
    fun updateTeacherInfo(uid: String,name : String,avatar: String, result: (UiState<String>) -> Unit)
}